docs/guide/login_and_registration/login_and_registration_cookbook.md

```diff
@@ -31,64 +31,61 @@ $token->setAttribute('customer_no', $customerNo);
 
 ### UserProvider
 
-UserProvider gets the created UsernamePasswordToken and his goal is to fetch the correct user from eZ.
+`UserProvider` gets the created `UsernamePasswordToken` and fetches the correct User from the content model.
 
-The default fetching functionality had to be extended because in the customer center user can be placed in different user groups (or without a group).
+The default fetching functionality is extended because in the Customer Center the User can be placed in different User Groups (or outside a group).
```

Comment:
```suggestion
The default fetching functionality is extended because, in the Customer Center, the User can be placed in different User Groups (or outside a group).
```